app/Http/Controllers/Assets/AssetCheckinController.php

```diff
@@ -75,6 +76,12 @@ public function store(AssetCheckinRequest $request, $assetId = null, $backto = n
 
         $this->authorize('checkin', $asset);
 
+        if(Gate::allows('audit',$asset)) {
+            if ($request->filled('log_audit') == "1") {
+                $this->authorize('audit', Asset::class);
```

Comment:
This duplicates the gate right above, and on failure will cause a forbidden error.

Reply:
I don't see a gate above it? only the `$this->authorize('checking',$asset);`
then everything else in the store method is just checking for the asset to exist?

Reply:
Line 79